schema_based_on_diagram.sql

```diff
@@ -0,0 +1,49 @@
+CREATE TABLE patients (
+    id SERIAL NOT NULL PRIMARY KEY,
+    name VARCHAR(60) NOT NULL,
+    date_of_birth DATE NOT NULL
+);
+
+CREATE TABLE medical_histories (
+    id SERIAL NOT NULL PRIMARY KEY,
+    patient_id INTEGER NOT NULL REFERENCES patients(id),
+    admitted_at timestamp NOT NULL,
+    status VARCHAR(60) NOT NULL
+);
+
+CREATE TABLE invoices(
+    id SERIAL NOT NULL PRIMARY KEY,
+    total_amount DECIMAL(10,2) NOT NULL,
+    generated_at timestamp NOT NULL,
+    payed_at timestamp NOT NULL,
+    medical_history_id INTEGER NOT NULL REFERENCES medical_histories(id)
+);
+
+CREATE TABLE invoice_items (
+    id SERIAL NOT NULL PRIMARY KEY,
+    unit_price DECIMAL(10,2) NOT NULL,
+    quantity INTEGER NOT NULL,
+    total_price DECIMAL(10,2) NOT NULL,
+    invoice_id INTEGER REFERENCES invoices(id),
+    treatment_id INTEGER REFERENCES treatments(id)
+);
+
+CREATE TABLE treatments (
+    id SERIAL NOT NULL PRIMARY KEY,
+    type VARCHAR(60),
+    name VARCHAR(60)
+);
+
+CREATE TABLE medical_histories_treatments (
+    PRIMARY KEY (medical_histories_id, treatments_id),
+    medical_histories_id INTEGER REFERENCES medical_histories(id),
+    treatments_id INTEGER REFERENCES treatments(id)
+);
+
+-- Indexes
+CREATE INDEX idx_patient_id ON medical_histories(patient_id);
+CREATE INDEX idx_medical_history_id ON invoices(medical_histories_id);
```

Comment:
- [ ] 👍 Great work on creating the index for the `medical_history_id` in the `invoices` table! However, there seems to be a small typo error in your code. You've used `medical_histories_id` instead of `medical_history_id`. To ensure accuracy, please consider updating your code to resolve this typo error. Keep up the good work! 👏🛠️

![image](https://github.com/ulises2607/Clinic-Database/assets/55840999/0ff6f263-225c-4396-b698-8b3b4a617425)